tooling/templatize/pkg/aks/kubeconfig.go

```diff
@@ -0,0 +1,61 @@
+package aks
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+)
+
+func GetKubeConfig(ctx context.Context, subscriptionID, resourceGroupName, aksClusterName string) (string, error) {
```

Comment:
If you're shelling out to `kubelogin`, any specific reason not to shell out to `az aks get-credential`?

Reply:
I see other parts of the code shelling out to `az` CLI as well - IMO save yourself the time and maintenance for this code if it's simple to just write `az` calls.

Reply:
i wanted to avoid shelling out as much as possible.
i wanted to revisite if we can use `kubelogin` as a library to avoid shelling out there

Reply:
i wrote down an action item